### ISSUE TITLE
High memory usage causing lag spikes (2023.4.19, Linux)

### ISSUE THREAD

**Issue by Memresable** [ppy/osu#23252]
### Type

Performance

### Bug description

The startup of the game became much slower, things aren't as smooth as before, due to memory usage being both really high on VRAM and usual RAM in the recent release (2023.4.19 on Linux) It became unplayable for me

I even commented on an Issue on what I thought caused this problem (I assumed on what caused it, no clue what is the actual cause) on this issue: https://github.com/ppy/osu-framework/pull/5508 but there weren't any comments on it after hours at the time of writing this

### Screenshots or videos

https://user-images.githubusercontent.com/86978624/233318177-1dd98a8f-d8c6-4bcb-9ff5-a0a4fb078d55.mp4

### Version

2023.4.19

### Logs

[database.log](https://github.com/ppy/osu/files/11284431/database.log)
[input.log](https://github.com/ppy/osu/files/11284432/input.log)
[network.log](https://github.com/ppy/osu/files/11284433/network.log)
[performance.log](https://github.com/ppy/osu/files/11284435/performance.log)
[runtime.log](https://github.com/ppy/osu/files/11284436/runtime.log)

**Comment by peppy** [comment]
Could you please try the legacy opengl renderer? Is there a reason you are not willing to?

**Comment by Memresable** [comment]
I tried the legacy one and it runs just fine there, also also there really wasn't a reason, I just wanted to try the latest renderers and not bothered with the legacy one

**Comment by frenzibyte** [comment]
If you're interested, can you give https://github.com/ppy/osu-framework/compare/master...frenzibyte:osu-framework:compute-mipmap-generation a test and see if it helps? See https://github.com/ppy/osu-framework/wiki/Testing-local-framework-checkout-with-other-projects for how to build osu! with local framework checkout. If you're not familiar with the process, then you can wait until the changes get in.

**Comment by qwert0p** [comment]
I've tested the game with that framework branch, it certainly works better, ram usage doesn't skyrocket like release, but it's still getting filled up slowly.

**Comment by frenzibyte** [comment]
Try decreasing `TextureStore.max_atlas_size` to 1024x1024 and see if that matches what you get on previous release.

**Comment by qwert0p** [comment]
It doesn't, previous release uses way less memory and it counts towards the process, current build somehow doesn't. Also when i decreased atlas size to 1024 it seems to crash when i open the map with veldrid exception and legacy renderer segfaults on start.

**Comment by frenzibyte** [comment]
Are you sure? It works perfectly fine for me.
```diff
diff --git a/osu.Framework/Graphics/Textures/TextureStore.cs b/osu.Framework/Graphics/Textures/TextureStore.cs
index 9d1d24d97..5812205d9 100644
--- a/osu.Framework/Graphics/Textures/TextureStore.cs
+++ b/osu.Framework/Graphics/Textures/TextureStore.cs
@@ -34,7 +34,7 @@ public class TextureStore : ITextureStore
 
         protected TextureAtlas Atlas;
 
-        private const int max_atlas_size = 4096;
+        private const int max_atlas_size = 1024;
 
         /// <summary>
         /// Decides at what resolution multiple this <see cref="TextureStore"/> is providing sprites at.
```

**Comment by qwert0p** [comment]
Okay, i hope I'm not dumb enough to mess up changing a number in the source code, there are the logs(running release mode, crashes even before the logo appears on debug) 
[runtime.log](https://github.com/ppy/osu/files/11290105/runtime.log)
[performance.log](https://github.com/ppy/osu/files/11290106/performance.log)
[network.log](https://github.com/ppy/osu/files/11290107/network.log)
[input.log](https://github.com/ppy/osu/files/11290108/input.log)
[database.log](https://github.com/ppy/osu/files/11290109/database.log)

**Comment by frenzibyte** [comment]
That's pretty weird, you'll have to investigate that yourself. Try attaching a breakpoint in `generateMipmapsViaComputeShader` and see which `CreateTextureView` call is failing and why so.

**Comment by frenzibyte** [comment]
Never mind, @qwert0p fetch latest changes and check again.

**Comment by qwert0p** [comment]
Yep, seems much better now, nothing seems to crash or behave weirdly and ram usage seems reasonable. I'll check everything tomorrow and report properly.

**Comment by CEO-Chair** [comment]
Seemingly same issue occurring on Windows 10 (Veldrid / OpenGL).

**Comment by frenzibyte** [comment]
There's no helpful information in your comment.

**Comment by tematikkp** [comment]
Not only that, but #23255

**Comment by Thesola10** [comment]
Is that the "shared memory" usage that didn't show up in a process or in-memory filesystem? I'm on Intel UHD btw, I don't know if there's a tool to track VRAM usage for that.
